src/main/java/org/folio/orders/utils/HelperUtils.java

```diff
@@ -299,14 +255,19 @@ private static double accumulate(Double a, Double b) {
 
   public static CompletableFuture<JsonObject> operateOnSubObj(HttpMethod operation, String url,
       HttpClientInterface httpClient, Context ctx, Map<String, String> okapiHeaders, Logger logger) {
+    return operateOnSubObj(operation, url, null, httpClient, ctx, okapiHeaders, logger);
+  }
+
+  public static CompletableFuture<JsonObject> operateOnSubObj(HttpMethod operation, String url, JsonObject body,
+      HttpClientInterface httpClient, Context ctx, Map<String, String> okapiHeaders, Logger logger) {
     CompletableFuture<JsonObject> future = new VertxCompletableFuture<>(ctx);
 
     logger.info("Calling {} {}", operation, url);
 
     try {
-      httpClient.request(operation, url, okapiHeaders)
+      httpClient.request(operation, (body != null ? body.toBuffer() : null), url, okapiHeaders)
         // In case there was failed attempt to delete order or particular PO line, the sub-objects might be already partially deleted.
-        .thenApply(HelperUtils::verifyAndExtractBodyIfFound)
+        .thenApply(response -> verifyAndExtractBody(operation, response))
```

Comment:
This is fairly minor, but I think this can be a little misleading... if I'm looking at the logs and run into a GET/404, I'm going to see that I got back and empty response, but in reality we probably got back a 404 with a text/plain response.  I think I'd rather see the GET/404 specific logic handled here instead of in a new variant of `verifyAndExtractBody(...)`.  You can then log an appropriate message saying you got back a 404 but you're ignoring it and continuing.

Reply:
Actually I was trying to handle `404` response a few lines below knowing that empty json object is for such case:
```java
if (json != null) {
  if (json.isEmpty()) {
    logger.warn("The {} {} operation completed with empty response body", operation, url);
```

I will try to make it more clear...